proposals/features-composite-dependencies.md

```diff
@@ -0,0 +1,117 @@
+
+### Direct curl
+
+We've seen instances where users directly `curl` a Feature's `install.sh` script to `bash`.  This strategy also doesn't merge in the other dev container configuration properties that a Feature may declare.
+
+### Additional inspiration
+
+Inspiration was taken from [this spec issue on the topic](https://github.com/devcontainers/spec/issues/109), the repositories listed above, [the buildpack specification](https://docs.cloudfoundry.org/buildpacks/understand-buildpacks.html), and [VS Code extension packs](https://code.visualstudio.com/blogs/2017/03/07/extension-pack-roundup).
+
+## Specification
+
+Introduce a new file type `devcontainer-feature.composite.json` with the following properties.
+
+| Property | Type | Description |
+|----------|------|-------------|
+| `id` | `string` | The ID of the Feature.  This follows the same semantics of the `id` property in the `devcontainer-feature.json` file. |
+| `version` | `string` | The version of the Feature.  This follows the same semantics of the `version` property in the `devcontainer-feature.json` file. |
+| `features` | `array` | An array of objects (in installation order) that define the Feature(s) that compose this Feature. |
+| `features.id` | `string` | The ID of the Feature that this Feature depends on. |
+| `features.version` | `string` | The version of the Feature that this Feature depends on. |
+| `features.detect` | `string` | A command that will be executed in a shell to determine if the Feature should be installed.  If the command returns a non-zero exit code, the Feature will be installed.  If the command returns a zero exit code, the remaining install steps will be skipped. |
+| `features.options` | `object` | An object of key/value pairs that will be passed to the Feature's `install.sh` script. |
+
+#### Example `devcontainer-feature.composite.json`
+
+```jsonc
+{
+    "id": "ghcr.io/devcontainers/features/composite",
+    "version": "1.0.0",
+    "features": [
+        {
+            "id": "ghcr.io/devcontainers/features/a", // Must be a standalone Feature. (A composite Feature cannot depend on a composite Feature).
+            "version": "1.2.3",  // An exact version is required.  We do not permit pinning to a major or minor version.
+            "detect": "a --version && cat /etc/a/.markerfile", // Only continue installation of this Feature if detect returns non-zero
+            "options": {
+                "bar": true
+            }
+        },
+        {
+            "id": "ghcr.io/microsoft/features/b",
+            "version": "sha256:45b23dee08af5e43a7fea6c4cf9c25ccf269ee113168c19722f87876677c5cb2", // SHA of the published artifact is OK
+            "detect": undefined, // Omit or set as 'undefined' to always install this Feature.
+            "options": {
+                "zip": "zap"
+            }
+        }
+    ]
+}
+```
+
+An optional `finalize.sh` script can be included, and will be fired after all Features have been installed.
```

Comment:
Do we have good examples for that?

Reply:
This could be omitted in the first pass if we find it to be unnecessary or without justification yet.  I'll leave for discussion

Reply:
I like that we are discussing various options. We should aim for as much simplicity as we can while still covering the important cases we have. It will be easier to later add to a simple design than a complex one.

Reply:
main